[Feat] #3 food search and allergy check

weejinyoung opened the pull request:
## feature 
- 페이지 별 5개 레시피 검색과 동시에 5개 레시피 각각의 알러지 유발 검사
- 사용자 알러지 수정, 읽기

## to do
- secure properties 를 마스터에서 삭제해야 함
- 로그인 시 사용자 이름을 response 에 추가해야 함
- 알러지 유발 식재료 정보를 추가해야 함
- openapi의 상세한 예외처리를 개발해야 함
- spring doc 으로 api 문서 배포 준비해야 함
- jenkins CI/CD 적용에 소요되는 시간을 고려해 github action 자동배포 적용을 우선해야 함
- controller 단에서 멤버 객체를 받아서 발생하는 트랜잭션 범위 밖 문제 논의해야 함
- 이번 feature 코드 리팩토링

weejinyoung committed:
[chore] add ignore

weejinyoung committed:
[feat] allergy properties setting

weejinyoung committed:
[feat] add enumset allergy attribute and converter

weejinyoung committed:
[feat] recipe openapi

weejinyoung committed:
[feat] resttemplate configuration

weejinyoung committed:
[feat] allergy service

weejinyoung committed:
[feat] allergy controller

weejinyoung committed:
[feat] allergy dto

weejinyoung committed:
[feat] openapi exception

daily1313 commented:
진영아 너무 늦었군.. 오늘 룸메이트랑 마무리 느낌으로 간단하게 한잔해서 이제야 리뷰 남기네 

깔끔하지 않은 OpenAPI로 개발하느라고 너무 고생했고 정말..

allergy.properties에 Map 형식으로 값을 저장하는 부분은 굉장히 신선한듯!

같이 팀플 합류해줘서 고맙고, 리뷰 바로 올릴테니까 시간 될 때 천천히 확인하고 내일 회의 때 길게 얘기해보자구

메서드 네이밍 룰 , 자바 코드 컨벤션, 코드 로직 개선 부분에 포커스 두고 리뷰 남길게

배포 때 까지 얼마 안 남았으니까 화이팅하구 팀원들끼리 같이 한강가서 힐링합시당~

daily1313 commented:
공통 리뷰 

Pull Request로 날린 여러 Class들에서 사용한 로그 확인용 @Slf4j annotation은 로그 출력 테스트가 완료됐으면 제거해줘도 될듯

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyType.java:
1. 이 클래스는 application 단에 두기 보다는 entity 클래스 만들고, entity에 두는게 맞다 생각하는데 어떤거 같아? 

2. AllergyProperties는 AllergyService class에서 DI를 해줬고, 72번째 줄에서 사용되어지고 있으므로 이것 까지는 Application에 두고 AllergyType만 Entity 패키지 파서 옮겨주면 좋을듯!

3. AllergyType 클래수 내부 첫 줄 개행 추가 (Java Code Convention)

weejinyoung replied:
1. 이 클래스는 application 단에 두기 보다는 entity 클래스 만들고, entity에 두는게 맞다 생각하는데 어떤거 같아?
a. 너가 말한 entity 클래스라는 게 정확히 어떤 건지 설명해줄 수 있어? 데이터베이스 모델링 상에서의 entity 를 말하는 건가?
2. AllergyProperties는 AllergyService class에서 DI를 해줬고, 72번째 줄에서 사용되어지고 있으므로 이것 까지는 Application에 두고 AllergyType만 Entity 패키지 파서 옮겨주면 좋을듯!
a. 1번 문항에 답이 된다면 패키지 구조도 확실하게 정할 수 있을 듯!
3. AllergyType 클래수 내부 첫 줄 개행 추가 (Java Code Convention)
a. 디테일 체크 구우욷
@daily1313

daily1313 replied:
1~2. 
쏘리 ㅠㅠ entity 클래스가 아니고 entity 패키지를 말하는거였으. 뒷말은 맞는 말인거 같아(db 모델링 상에서 entity를 의미하지). 근데, 실질적으로 rdb table에 매핑되는 entity가 아니니까 application에 둬도 될거 같아. 이거는 디코로 조금 더 얘기해보자! 너무 내가 패키징에 포커스를 둔거 같아 

회의 후 확실하게 결정해보는걸루 하자 진영아!

weejinyoung replied:
entity packaging 으로 결정!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+
+    @Transactional
+    public GetMyAllergiesResponse changeAllergies(Member member, ChangeAllergiesRequest request) {
+        Member curretnMember = memberRepository.findByUsername(member.getUsername()).orElseThrow(MemberNotFoundException::new);
```
네이밍 curretnMember -> currentMember로 변경하면 될듯!

weejinyoung replied:
오우 실수체크 고맙다아

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+    private final AllergyProperties allergyProperties;
+    private final MemberRepository memberRepository;
+
+    public CheckAllergyResponse checkAllergy(Member member, int page, String recipeName) {
```
메서드 네이밍 checkAllergyByRecipeName 어떨까? (쿼리파라미터로 recipeName을 넘겨주니까)

weejinyoung replied:
이전 플젝에서 어떤 일이 있었냐면

`getMember(String name)` 과 `getMemberByName(String name)` 이 있을 때 전자를 선택하는 컨벤션을 정하게 됐었어
나는 가독성 측면에선 `getMemberByName(String name)` 이 좋다고 생각했었는데 팀원은 매개변수의 이름까지 메소드의 의미를 전달하는 데에 포함될 수 있지 않겠냐는 생각을 하더라고
매개변수애 `String name` 이 있다면 메소드 이름엔 굳이 byName 을 포함시키지 않아도 된다는 주의였던 거지

그래서 이전 플젝에선 by 를 거의 사용하지 않았고 그게 습관이 돼서 이번에도 그렇게 썼나봐

checkAllergyByRecipeName 으로 가는 거 나도 좋다고 생각해 바로 적용할게
특히 `checkAllergy(Member member, int page, String recipeName)` 메소드는 page 라는 매개변수도 있으니까 recipeName 으로 알러지를 체크한다는 의미를 명확하게 표현할 수 있어서 더 좋은 것 같아
위에서 했던 얘기는 너가 나중에 다른 팀에서 네이밍에 대한 컨벤션을 정할 때 참고하면 좋을 것 같아서 해봐따~

daily1313 replied:
네이밍은 초창기에 미리 같이 논의했어야 하는데..ㅎ 경험까지 얘기해서 이해하기 좋다

나중에 플젝할때도 또 고민해보께 굿굿!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+            Set<String> ingredients = convertStringToSet(recipe.getPartsDetails());
+            Pair<Set<String>, Set<String>> ingredientsAndCauses = findCauses(member.getAllergies(), ingredients);
+            CheckedFood checkedFood = CheckedFood.createCheckedFood(name, imageUrl, ingredientsAndCauses.getFirst(), ingredientsAndCauses.getSecond());
+            checkAllergyResponse.addResult(checkedFood);
+        }
+        return checkAllergyResponse;
```
- 이부분은 뒤에 checkedFood 관련해서 얘기하면서 같이 얘기하면 좋을 듯!

weejinyoung replied:
https://github.com/2023ICE/backend/pull/4#discussion_r1391529322
해결!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+    @Transactional
+    public GetMyAllergiesResponse changeAllergies(Member member, ChangeAllergiesRequest request) {
+        Member curretnMember = memberRepository.findByUsername(member.getUsername()).orElseThrow(MemberNotFoundException::new);
+        EnumSet<AllergyType> allergies = EnumSet.noneOf(AllergyType.class);
+        for (String allergy : request.getAllergies()) {
+            allergies.add(AllergyType.nameOf(allergy));
+        }
+        EnumSet<AllergyType> newAllergies = curretnMember.changeAllergies(allergies);
+        return GetMyAllergiesResponse.toDto(
+                newAllergies.stream()
+                        .map(AllergyType::getName)
+                        .toList());
+    }
```
Enumset을 새로만들고 알러지 정보를 가져와서 하나씩 allergy를 추가하는 로직이 보이는데 요거는 살짝 (change)update랑 무관하다고 생각해 

조회를 해서 업데이트를 하는게 맞는거 같은데, 바로 아래 메서드 처럼 가는게 어떨까? 

`EnumSet<AllergyType> allergies = member.getAllergies();`

너 생각은 어때 진영아

weejinyoung replied:
1. 우리의 서비스 플로우 상 내 알러지 정보를 가져오고, 바꾼 알러지 정보를 보내는 두 API 만으로 내 알러지 정보를 다뤄야 해

2. 우리는 알러지를 유발하는 재료의 정보를 RDB 에 저장하는 방식이 아닌 서버 메모리에 직접 올려놓는 방식을 선택했고
이에 따라 Member Entity Class 에는 EnumSet 타입으로 선언한 attribute 로, member table 에는 string 타입의 column 으로 사용자의 알러지 정보를 저장하게 됐어
만약 EnumSet<AllergyType> ["갑각류", "밀"] 에서 ["갑각류", "유제품"] 으로 알러지 정보가 바뀌어야 하는 상황에서 기존의 update 방식을 사용해야 한다면 ["갑각류", "밀"] 과 ["갑각류", "유제품"] 의 요소들을 하나하나 대조하고 수정 삭제해야 하기 때문에 로직이 꽤 복잡해지고 예외상황도 많이 발생할 거야

위와 같은 이유들로 기존의 방식처럼 CRUD 를 구분하는 방법은 어려울 수 밖에 없지
그래서 난 update 스러운 PATCH 보단 자원이 전부 교체되는 PUT 의 매커니즘으로 접근해서 CRUD 가 아닌 CR 두 방식으로만 사용자의 알러지 정보를 다루기로 했고
request 의 알러지 정보 필드를 담은 새로운 EnumSet 변수를 만들고 기존의 Member Entity 필드인 `EnumSet<AllergyType> allergies` 와 교체하는 방법을 사용했어

이따 디코로 더 깊게 이야기해보자 혹시 더 나은 방법이 있다면 말해조!
@daily1313

daily1313 replied:
내부 로직이 복잡해진다는 이유를 간과했던거 같네.. (사실 이거는 이렇게 로직을 구성한 이유가 있을 거 같았는데, 너무 내가 단편적으로 Update에 포커스를 둔 거 같았네)

너 코멘트를 읽고 나니까 Update지만, Create 스럽게 로직을 구성하는게 여러 방면에서 더욱 깔끔할거 같다. (앞서 설명한 예외상황, Allergy 비교에 의한 시간 복잡도 측면들을 고려했을 때)

이거는 너가 로직 짜놨던 방식대로 가는게 훨씬 효율적일 것 같아! 굿 ㅎㅎ

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+        }
+        return list;
+    }
+
+
+
```
개행 제거 부탁해!

weejinyoung replied:
디테일 체크 구욷

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+        Set<String> causes = new LinkedHashSet<>();
+        Set<String> checkedIngredients = new LinkedHashSet<>();
+        Map<String, AllergyType> allergenicIngredients = allergyProperties.getAllergenicIngredients();
+        for (String ingredient : ingredients) {
+            if (allergenicIngredients.containsKey(ingredient)) {
+                AllergyType allergy = allergenicIngredients.get(ingredient);
+                if (allergies.contains(allergy)) {
+                    checkedIngredients.add(ingredient);
+                    causes.add(allergy.getName());
+                }
+            }
+        }
```
for문 안에 if문 메서드 추출해서 indent 줄여보면 괜찮을거 같아

weejinyoung replied:
아이디어 땡큐!
리팩한 거 확인해줘!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyType.java:
```diff
@@ -0,0 +1,29 @@
+
+    private final String name;
+
+    public static AllergyType nameOf(String name) {
+        for (AllergyType allergyType : AllergyType.values()) {
+            if (allergyType.getName().equals(name)) {
+                return allergyType;
+            }
+        }
+        return null;
+    }
```
null을 반환하기 보다 NPE(Null Point Exception)를 방지하기 위해 Optional과 같은 값을 반환하는거는 어떻게 생각해? 두 가지 방안이 있다고 생각해

1. Optional 

```text
public static Optional<AllergyType> nameOf(String name) {
    return Arrays.stream(AllergyType.values())
            .filter(allergyType -> allergyType.getName().equals(name))
            .findFirst();
}
```
2. Exception 터트려주기
```text
public static AllergyType nameOf(String name) {
    return Arrays.stream(AllergyType.values())
            .filter(allergyType -> allergyType.getName().equals(name))
            .findFirst()
            .orElseThrow(InvalidAllergyTypeNameException.class));
}
```

weejinyoung replied:
옛날에 허겁지겁 하던 프젝 코드를 그냥 가져온 걸 들켰구만
덕분에 반성할 수 있었다구
Exception 상황이 더 맞는 것 같다!
로직까지 짜줘서 고맙다!!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyTypeSetConverter.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Override
+    public String convertToDatabaseColumn(EnumSet<AllergyType> attribute) {
+        if (attribute == null || attribute.equals(EnumSet.noneOf(AllergyType.class))){
+            return null;
```
이부분은 개인적으로 DB에 저장하는 부분이니까 Null로 가는게 맞다고 생각함 

이유: [https://dont-think-about-too-much.github.io/2021/06/03/think-about-missing-value/](url)

weejinyoung replied:
나 이거 null 로 저장한 것 같은데 너도 이렇게 하는 게 좋다는 얘기지?
@daily1313

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyTypeSetConverter.java:
```diff
@@ -0,0 +1,35 @@
+            return null;
+        }
+        StringBuilder sb = new StringBuilder();
+        attribute.forEach(e -> sb.append(e.name()).append(","));
+        String result = sb.toString();
+        if(result.charAt(result.length() - 1) == ',') result = result.substring(0, result.length() - 1);
+        return result;
```
로직을 봤을 때, 마지막에 콤마(,)가 무조건 나오니까 if문으로 검사 안하고 바로 마지막  요소 제거해주고 반환을 하면 될듯해 

1. result.substring(0, result.length() -1) 로 가도 괜찮을 거 같고  (String)
2. sb.deleteCharAt(sb.length() - 1)로 String 변환 전에 StringBuilder에서 제거 (StringBuilder)

weejinyoung replied:
무작정 가져온 코드의 폐해다..
고마워 바로 적용할게!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyTypeSetConverter.java:
되게 사소한건데, if문 적용할 때 {}(중괄호) 넣어주면 좋을 것 같아. (Convention)

weejinyoung replied:
아아 코드를 가져올 때 체크했어야 하는데 땡큐

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/dto/ChangeAllergiesRequest.java:
```diff
@@ -0,0 +1,14 @@
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ChangeAllergiesRequest {
+    private List<String> allergies;
+}
```
클래스 내부 첫줄 개행 추가 부탁행

weejinyoung replied:
디테일 체크 구욷

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/presentation/AllergyController.java:
```diff
@@ -0,0 +1,46 @@
+
+    private final AllergyService allergyService;
+
+    @GetMapping("/check")
+    public ResponseEntity<CheckAllergyResponse> checkAllergy(@AuthMember Member member, int page, String recipeName) {
```
네이밍 마찬가지로 checkAllergyByRecipeName으로!

weejinyoung replied:
https://github.com/2023ICE/backend/pull/4#discussion_r1391491764
ditto!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/presentation/AllergyController.java:
```diff
@@ -0,0 +1,46 @@
+        GetMyAllergiesResponse response = allergyService.changeAllergies(member, request);
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(response);
+    }
+
```
개행처리 부탁할게!

weejinyoung replied:
디테일 체크 구욷

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/presentation/dto/CheckedFood.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.Set;
+
+@AllArgsConstructor
+@Data
+public class CheckedFood {
+    private String name;
```
1. 클래스 내부 개행처리 
2. Dto에서도 @Setter를 지양하면 좋으니 @Data 보다 @Getter로 가면 좋을거 같아(사실 Dto라서 @Setter가 있어도 무관하다고 생각하는데 개인적으로 쓰이는 경우가 없으면 빼도 될거 같아)
3. Dto Class임을 명시적으로 드러내기 위해 class naming을 CheckedFoodResponse로 하면 어떨까?

weejinyoung replied:
오옹 내가 미처 하지 못했던 좋은 생각인 것 같다
이건 바로 적용할게!

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/presentation/dto/GetMyAllergiesResponse.java:
```diff
@@ -0,0 +1,15 @@
+import java.util.List;
+
+@AllArgsConstructor
+@Data
+public class GetMyAllergiesResponse {
+    private List<String> allergies;
```
클래스 내부 첫 줄 개행처리 부탁해

weejinyoung replied:
디테일 체크 구욷

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/presentation/dto/CheckAllergyResponse.java:
```diff
@@ -0,0 +1,22 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+public class CheckAllergyResponse {
+
+   private List<CheckedFood> result = new ArrayList<>();
+
+   public void addResult(CheckedFood checkedFood) {
+      result.add(checkedFood);
+   }
```
1. Dto 자체가 Data Transfer(데이터 전달)이 주 목적인데, result.add() 메서드가 있어도 되지는지 궁금해 너 생각은 어때?
2. new ArrayList<>()를 여기 작성하기 보다 Service Class에서 생성해주는게 깔끔하다고 생각함
3. toDto 느낌으로 List<CheckedFood>를 넘겨주면 좋을거 같아

weejinyoung replied:
오 맞아
오랜만에 코딩하다보니까 알고리즘 푸는 로직을 짜버렸네..ㅎ
땡큐 바로 적용할게

daily1313 replied:
ㅎㅎ 알고리즘 하다가 스프링하면 그럴 수 있다 생각함! 굿굿

daily1313 commented on src/main/java/allercheck/backend/domain/member/entity/Member.java:
```diff
@@ -85,4 +92,9 @@ private static void validateNameFormat(final String name) {
         }
     }
+
+    public EnumSet<AllergyType> changeAllergies(EnumSet<AllergyType> allergies) {
+        this.allergies = allergies;
+        return this.allergies;
+    }
```
1. 나도 알러지 바꾸는 부분은 여기서 하는게 맞다고 생각하고 잘한듯. return 타입을 void로 하면 어떨까?
2. AllergyService에서 안 끌고와도 괜찮을거 같아
3. Entity, Service Layer에서는 각각 다음과 같이 작성해주면 될듯해


```text
Member.class 

currentMember.changeAllergies(allergies);

AllergySerivce

public void changeAllergies(EnumSet<AllergyType> allergies {
	this.allergies = allergies;
}
```

weejinyoung replied:
아 나도 이건 고민 좀 했던 건데
프론트 친구들한테 어떤 reponse 가 가는 게 좋은지 한번 물어보는 거 어때?
@daily1313

daily1313 replied:
오키! 이거 수정하는 부분은 현의한테 물어봐서 어떤걸 원하는지 얘기해보자고

weejinyoung replied:
void 리턴으로 결정!

daily1313 commented on src/main/java/allercheck/backend/domain/openapi/application/RecipeOpenApiService.java:
```diff
@@ -0,0 +1,52 @@
+    private final RestTemplate restTemplate;
+
+    public GetRecipeResponse getRecipeResponse(final int page, final String recipeName) {
+        int startIdx = (page - 1) * 5;
+        int endInx = page * 5;
+        URI uri = UriComponentsBuilder
+                .fromUriString(url)
+                .path("/{secret}/COOKRCP01/json/{startIdx}/{endIdx}/{recipeName}")
```
endInx -> endIdx로 수정 부탁해

weejinyoung replied:
실수체크 고맙다아아

daily1313 commented on src/main/java/allercheck/backend/domain/openapi/application/RecipeOpenApiService.java:
```diff
@@ -0,0 +1,52 @@
+                .expand(secret, startIdx, endInx, "RCP_NM=" + recipeName)
+                .toUri();
+        ResponseEntity<GetRecipeResponse> response = restTemplate.getForEntity(uri, GetRecipeResponse.class);
+        if (response.getStatusCode().is2xxSuccessful()) {
+            return response.getBody();
+        } else {
+            throw new OpenApiConnectionFailureException();
+        }
```
else 예약어 안쓰고 바로 예외 터트리면 될 거 같아!

weejinyoung replied:
아이디어 땡큐!
이 부분이 구현에만 집중했던 부분이라서 꽤 많은 리팩이 적용될 거야
리팩한 거 확인해줘!

daily1313 replied:
천천히 리팩하고 새벽 늦게까지 comment 다느라고 고생했어!

daily1313 commented on src/main/java/allercheck/backend/domain/openapi/application/RecipeOpenApiService.java:
```diff
@@ -0,0 +1,52 @@
+            throw new OpenApiConnectionFailureException();
+        }
+    }
+
+
+
+
+
+
+
```
개행 제거 부탁해!

weejinyoung replied:
디테일 체크 굳

daily1313 commented on src/main/java/allercheck/backend/domain/openapi/presentaion/GetRecipeResponse.java:
GteRecipeResponse Class Name을 RecipeResponse로 바꿔주면 어떨까? Response 자체가 Get 느낌이 있는거 같아서!

weejinyoung replied:
요거 좋은 생각인 것 같다 바로 적용해볼게

daily1313 commented:
To Do List 관련 Comment

1. secure properties 를 마스터에서 삭제해야 함 (.gitignore로 *.properties 처리해줬으니 해결) 
2. 로그인 시 사용자 이름을 response 에 추가해야 함 (dto -> name 필드 추가)
3. 알러지 유발 식재료 정보를 추가해야 함 (팀원들과 회의 후, 개별 조사 -> 노션 확인 완료)
4. openapi의 상세한 예외처리를 개발해야 함 (Client(Front-end) 입장에서 가장 중요한 부분 제일 먼저일듯)
5. spring doc 으로 api 문서 배포 준비해야 함 (API가 몇 개 되지 않아, 기존 API 명세서를 강하게 보완하는 방향으로 가보고, 프로젝트가 끝난 후 같이 스터디를 하는 쪽으로 가는 것이 좋을 듯!)
6. jenkins CI/CD 적용에 소요되는 시간을 고려해 github action 자동배포 적용을 우선해야 함 (확인)
7. controller 단에서 멤버 객체를 받아서 발생하는 트랜잭션 범위 밖 문제 논의해야 함 (Auth 관련 로직 문제일 가능성이 100%, Member 객체를 받는 것이 아닌 , 추후 Member Id 값을 받도록 코드 개선)
8. 이번 feature 코드 리팩토링 (배포까지 아직 여유가 있어 deadline이 남은 상태, feature 코드 리팩토링 바로 진행하면 될듯)

daily1313 commented on src/main/java/allercheck/backend/domain/allergy/application/AllergyService.java:
```diff
@@ -0,0 +1,101 @@
+                        .toList());
+    }
+
+    private Pair<Set<String>, Set<String>> findCauses(EnumSet<AllergyType> allergies, Set<String> ingredients) {
```
findCausesByIngredients 네이밍 변경하면 더 괜찮을거 같아

weejinyoung replied:
https://github.com/2023ICE/backend/pull/4#discussion_r1391491764
ditto!

daily1313 reviewed (commented):
확인 한번 부탁해! 진영아 ~!

weejinyoung committed:
[refactor] AllergyType, AllergyProperties

1. entity 의 의미가 데이터베이스 모델링에 국한되지 않는다는 판단 하에 entity packaging
2. AllergyType.nameOf 로직 개선
3. AllergyProperties 에 getAllergyByIngredient, isAllergenicIngredient 메소드 추가

weejinyoung committed:
[refactor] AllergyTypeSerConverter

1. 항상 마지막 문자를 제거하도록 convertToDatabaseColumn 메소드 로직 개선

weejinyoung committed:
[refactor] GetRecipeResonse

1. GetRecipeResonse -> RecipeOpenApiResponse 로 네이밍

weejinyoung committed:
[refactor] CheckedFood

1. CheckedFood -> CheckRecipeResponse 로 네이밍
2. createCheckedFood -> toDto 로 네이밍

weejinyoung committed:
[refactor] CheckAllergyResponse

1. addResult -> toDto 로 네이밍

weejinyoung committed:
[refactor] AllergyService, RecipeOpenApiService

1. RecipeService interface 와 Recipe 를 사용해 AllergyService 와 RecipeOpenApiService 의 결합도를 낮춤
2. convertIngredientsStringToSet 메소드를 AllergyService 에서 RecipeOpenApiService 로 옮겨 응집도를 높임
3. AllergyService 의 로직, 네이밍 개선
4. RecipeOpenApiService 의 상수값들 전역으로 관리

weejinyoung committed:
[refactor] etc

1. AllergyController 의 메소드 네이밍
2. 개행

weejinyoung committed:
[refactor] RecipeOpenApiService

1. checkRecipe -> validateRecipeOpenApiResponse 네이밍

weejinyoung committed:
[refactor] GetMyAllergiesResponse

1. GetMyAllergiesResponse -> MyAllergiesResponse 네이밍

weejinyoung committed:
[refactor] AllergyTypeSetConverter

1. if 문 중괄호 추가

weejinyoung committed:
[refactor] etc

weejinyoung committed:
[refactor] changeAllergies

1. void 를 return 하는 것으로 변경

weejinyoung committed:
[refactor] signIn

1. 응답에 사용자의 이름 추가

weejinyoung committed:
[refactor] AllergyType

1. 달걀 -> 난류

weejinyoung commented:
단 두번의 풀 리퀘스트로 기능 개발이 끝났네
범이 일 끝나고 와서 개발하느라, 꼼꼼하게 리뷰하느라 고생 너무 많았다
배포까지 조금만 더 화이팅

weejinyoung committed:
[refactor] AllergyController

1. String -> Void

weejinyoung committed:
[refactor] SignUpResponse

1. SignUpResponse -> SignInResponse

daily1313 commented:
고생햇슈~
